### PR TITLE
`Development`: Add course messages e2e tests

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

After adding e2e tests for the communication feature within Artemis, we now add e2e tests for the messages functionality. 

### Description

This PR adds basic channel tests for the course messages functionality within Artemis. For now we have the following tests:
- Create announcement channel
- Create unrestricted channel
- Create public channel
- Create private channel
- Check for not allowed channel names
- Edit channel
- Join channel
- Leave channel

In a followup PR we will add tests for writing messages, creating group chats, 1 to 1 chats etc.

#### Code Review
- [ ] Review 1
- [ ] Review 2

### CODE REVIEW

**Review by simon-hng — DISMISSED** [review]
Code changes look good and tests look meaningful

File: src/test/cypress/e2e/course/CourseMessages.cy.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                courseMessages.leaveChannel(channel.id!);
+                courseMessages.checkBadgeJoined(channel.id!).should('not.exist');
+            });
+        });
+    });
+
+    after('Delete Course', () => {
+        cy.login(admin);
+        if (courseId) {
+            //courseManagementRequest.deleteCourse(courseId).its('status').should('eq', 200);
```
Comment (danielbode) [inline src/test/cypress/e2e/course/CourseMessages.cy.ts:149]:
```suggestion
            courseManagementRequest.deleteCourse(courseId).its('status').should('eq', 200);
```

**Review by danielbode — APPROVED** [review]
Tests and code changes look good

**Review by simon-hng — APPROVED** [review]
Reapprove after removing dead code

**Pull request merged**